src/ledger/backend/writeconn.ts

```diff
@@ -68,56 +68,26 @@ export class LedgerWriteConn extends LedgerReadConn implements LedgerWriter {
 			calls.push(...depositCalls);
 		}
 
-		return this.call(calls);
+		return { stages: this.call(calls), numStages: calls.length };
 	}
 
-	private async call(
+	private async *call(
 		calls: DepositCalls,
```

Comment:
I think the type `DepositCalls` needs to be renamed. It's not only `DepositCalls` but calls in general, right?

Reply:
Same for type `DepositCall` in file `tokenmanager.ts`. Looks like a generic `Call` to me.